src/NuGet.Core/NuGet.ProjectModel/PackageSpecExtensions.cs

```diff
@@ -43,5 +44,31 @@ public static ProjectRestoreMetadataFrameworkInfo GetRestoreMetadataFramework(th
 
             return frameworkInfo ?? new ProjectRestoreMetadataFrameworkInfo();
         }
+
+        /// <summary>
+        /// Merge the CentralVersion information to the package reference information.
+        /// </summary>
+        /// <param name="project">The package spec to update.</param>
+        /// <returns>The updated package spec.</returns>
+        public static PackageSpec WithCentralVersionInformation(this PackageSpec project)
```

Comment:
The name here suggests a pattern where a new PackageSpec is returned, but that's not quite what's happening. 

The method has side-effects to the original PackageSpec. 

I think I would rather change the name. I wouldn't want us to clone.

Reply:
Renamed to ApplyCentralVersionInformation.

Regarding the performance numbers - there was an almost 20% improvement observed from a baseline to a run including more perf improvements. Among them was clone removal. 
The 20% improvement was for an E2E scenario experiment  running msbuild /t:restore

For the same scenarios the projectRestoreInformation telementry data shows almost 50% improvement

![image](https://user-images.githubusercontent.com/16580006/93394501-2bc68700-f829-11ea-863e-3c3d097a8305.png)